### ISSUE TITLE
[Bug] chained parameters search returns all Resources

### ISSUE THREAD

**Issue by lquanx** [awslabs/fhir-works-on-aws-search-es#138]
**Describe the bug**
For API request `GET DetectedIssue?patient:Patient.name=aaa` . I got all DetectedIssue Resource with patient fields

**To Reproduce**
1. Git clone smart-mainline branch ` git clone -b smart-mainline git@github.com:awslabs/fhir-works-on-aws-deployment.git`
2. Deploy AWS FHIR Works
3. Create Patient and DetectedIssue Resources.
```
POST /Patient
{
    "resourceType": "Patient",
    "name": [
        {
            "family": "Chalmers"
        }
    ]
}
```
```
POST /DetectedIssue
{
    "resourceType": "DetectedIssue",
    "patient": {
        "reference": "Patient/{{Patient ID from the above API request}}"
    }
}
```
4. Search DetectedIssue with Patient with any name.
```
GET /DetectedIssue?patient:Patient.name=fdsafds
```
Search DetectedIssue with Patient with empty name
```
GET /DetectedIssue?patient:Patient.name=
```

**Expected behavior**
Empty list of DetectedIssue is returned

**Actual behavior**
A DetectedIssue is returned

**Comment by lquanx** [comment]
Thanks for adding Chained Parameters search. I'm not sure why all resources are returned when I use the chained parameter search though. Thanks for taking a look into the issue.

**Comment by ssvegaraju** [comment]
Hi, thanks for bringing up this issue! I was able to reproduce the behavior described above, and have added this to our backlog. Also, FYI, the behavior with the empty name parameter returning all the resources is expected. However the behavior of a name that doesn't exist returning all of the resources appears to be faulty.

**Comment by lquanx** [comment]
got it. Thanks

**Comment by ssvegaraju** [comment]
Hi @Lin-002, we just published a fix for this in version 3.9.1, thanks again for bringing up this issue. I'm going to close this issue for now, but please feel free to reopen if there's anything else!
